crates/cubecl-common/src/future.rs

```diff
@@ -3,12 +3,17 @@ use core::future::Future;
 
 /// Block until the [future](Future) is completed and returns the result.
 pub fn block_on<O>(fut: impl Future<Output = O>) -> O {
-    #[cfg(not(feature = "std"))]
+    #[cfg(target_family = "wasm")]
+    {
+        super::reader::read_sync(fut)
+    }
+
+    #[cfg(all(not(target_family = "wasm"), not(feature = "std")))]
     {
         embassy_futures::block_on(fut)
     }
 
-    #[cfg(feature = "std")]
+    #[cfg(all(not(target_family = "wasm"), feature = "std"))]
     {
         futures_lite::future::block_on(fut)
```

Comment:
Maybe worth switching all the block_ons to embassy if we're using it

Reply:
When `std`, we're using `futures-lite` instead.

Reply:
Ok but... why :D Seems better to keep one async runtime, even if the runtime is just all block_on's. It's not a big difference though.

Reply:
We can't catch unwind with embassy, which is great for autotune. Also we don't pull the runtime for embassy, it's just a loop with a busy waiting.